extensions/ql-vscode/src/extension.ts

```diff
@@ -906,6 +906,7 @@ async function activateWithInstalledDistribution(
       variantAnalysis: VariantAnalysis,
       token: CancellationToken
     ) => {
+      variantAnalysisManager.variantAnalysisAddedEventEmitter.fire({ variantAnalysis });
```

Comment:
I'm not sure whether it makes sense to emit this event in the command, rather than in a method of the variant analysis manager. I would expect this command to also be called when VSCode is restarted and that should probably not emit this event (the variant analysis should have been added to the query history already at that point).

I think it would be clearer to fire this event from within `run-remote-query.ts` by calling a method on the variant analysis manager.

Reply:
Thanks for the advice, I hadn't taken into account what happens when VSCode is restarted. Have moved the event firing into `run-remote-query.ts` as you suggested.